near/omni-tests/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+
+        let prover_contract = worker.dev_deploy(&std::fs::read(MOCK_PROVER_PATH)?).await?;
+        // Deploy and init locker
+        let locker_contract = worker.dev_deploy(&std::fs::read(LOCKER_PATH)?).await?;
+        locker_contract
+            .call("new")
+            .args_json(json!({
+                "prover_account": prover_contract.id(),
+                "mpc_signer": "mpc.testnet",
+                "nonce": U128(0)
+            }))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+
+        // Create relayer account
+        let relayer_account = worker
+            .create_tla(relayer_account_id(), worker.dev_generate().await.1)
+            .await?
+            .unwrap();
+
+        // Storage deposit and transfer tokens
+        token_contract
+            .call("storage_deposit")
+            .args_json(json!({
+                "account_id": locker_contract.id(),
+                "registration_only": true,
+            }))
+            .deposit(NEP141_DEPOSIT)
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+
+        token_contract
+            .call("ft_transfer")
+            .args_json(json!({
+                "receiver_id": locker_contract.id(),
+                "amount": U128(amount),
+            }))
+            .deposit(NearToken::from_yoctonear(1))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+
+        // Add factory address
+        locker_contract
+            .call("add_factory")
+            .args_json(json!({
+                "address": eth_factory_address(),
+            }))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+
+        // Fin transfer
+        relayer_account
+            .call(locker_contract.id(), "fin_transfer")
+            .args_borsh(FinTransferArgs {
+                chain_kind: omni_types::ChainKind::Eth,
+                storage_deposit_args: StorageDepositArgs {
+                    token: token_contract.id().clone(),
+                    accounts: storage_deposit_accounts,
+                },
+                prover_args: borsh::to_vec(&ProverResult::InitTransfer(InitTransferMessage {
+                    emitter_address: eth_factory_address(),
+                    transfer: TransferMessage {
+                        origin_nonce: U128(1),
+                        token: token_contract.id().clone(),
+                        recipient: OmniAddress::Near(account_1().to_string()),
+                        amount: U128(amount - fee),
+                        fee: U128(fee),
+                        sender: eth_eoa_address(),
+                    },
+                }))
+                .unwrap(),
+            })
+            .deposit(NEP141_DEPOSIT.saturating_mul(2))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+
```

Comment:
We can also verify that the balance has changed correctly.

Reply:
418842eaaf5cb6a9e5337af071e20a8271d2e808